src/renderers/rust/templates/instructionsCpiPageBuilder.njk

```diff
@@ -0,0 +1,67 @@
+/// `{{ instruction.name | snakeCase }}` CPI instruction builder.
+pub struct {{ instruction.name | pascalCase }}Builder<'a> {
+  program: &'a solana_program::account_info::AccountInfo<'a>,
+{% for account in instruction.accounts %}
+  {% if account.isSigner === 'either' %}
+    {{ account.name | snakeCase }}: Option<(&'a solana_program::account_info::AccountInfo<'a>, bool)>,
+  {% else %}
+    {{ account.name | snakeCase }}: Option<&'a solana_program::account_info::AccountInfo<'a>>,
+  {% endif %}
+{% endfor %}
+{% for arg in instructionArgs %}
+  {% if not arg.default %}
+    {{ arg.name | snakeCase }}: {{ arg.type if arg.innerOptionType else 'Option<' + arg.type + '>' }},
+  {% endif %}
+{% endfor %}
+}
+
+impl<'a> {{ instruction.name | pascalCase }}Builder<'a> {
+  pub fn new(program: &'a solana_program::account_info::AccountInfo<'a>) -> Self {
```

Comment:
```suggestion
impl<'a, 'info> {{ instruction.name | pascalCase }}Builder<'a, 'info> {
  pub fn new(program: &'a solana_program::account_info::AccountInfo<'info>) -> Self {
```

Making the `AccountInfo` reference lifetime as long as the `AccountInfo`'s data buffer lifetime results in a lifetime error:

```
error: lifetime may not live long enough
 --> onchain-consumer/src/lib.rs:8:23
  |
7 | pub fn etp(program_id: &Pubkey, accounts: &[AccountInfo], instruction_data: &[u8]) -> ProgramResult {
  |                                 --------  - let's call the lifetime of this reference `'1`
  |                                 |
  |                                 has type `&[AccountInfo<'2>]`
8 |     let mut builder = AddConfigLinesBuilder::new(&accounts[0]);
  |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ argument requires that `'1` must outlive `'2`
```

We need to introduce a separate lifetime for the reference

Need to change this for all the other `&'a AccountInfo<'a>`s too

Reply:
Actually, nvm, this build error goes away if you change the entrypoint fn signature to `pub fn etp<'a>(program_id: &Pubkey, accounts: &'a [AccountInfo<'a>], instruction_data: &[u8]) -> ProgramResult`. I guess it's up to you guys to decide if constraining the reference 2 lifetimes to the same is too constraining.